iceoryx_hoofs/test/moduletests/test_vocabulary_variant.cpp

```diff
@@ -352,6 +352,42 @@ TEST_F(variant_Test, MoveCTorWithoutValueResultsInInvalidVariant)
     ASSERT_THAT(ignatz.index(), Eq(iox::INVALID_VARIANT_INDEX));
 }
 
+TEST_F(variant_Test, MoveCTorWithVariantLeadToSameValue)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "dc2a2aff-1fcd-4679-9bfc-b2fb4d2ae928");
+    iox::variant<int, float, ComplexClass> schlomo;
+    schlomo = ComplexClass(2, 3.14F);
+    iox::variant<int, float, ComplexClass> ignatz(std::move(schlomo));
+    ASSERT_THAT(schlomo.index(), Eq(iox::INVALID_VARIANT_INDEX));
```

Comment:
```suggestion
    // NOLINTJUSTIFICATION check if move is invalidating the object
    // NOLINTNEXTLINE(bugprone-use-after-move,hicpp-invalid-access-moved,clang-analyzer-cplusplus.Move)
    ASSERT_THAT(schlomo.index(), Eq(iox::INVALID_VARIANT_INDEX));
```
Please add that clang-tidy suppression to all new tests that access the moved from object after the move.

You can also run clang-tidy locally by
```sh
tools/scripts/clang_tidy_check.sh scan_list .clang-tidy-diff-scans.txt iceoryx_hoofs/test/moduletests/test_vocabulary_variant.cpp
```

Reply:
> Please add that clang-tidy suppression to all new tests that access the moved from object after the move.
> 
> You can also run clang-tidy locally by
> 
> ```shell
> tools/scripts/clang_tidy_check.sh scan_list .clang-tidy-diff-scans.txt iceoryx_hoofs/test/moduletests/test_vocabulary_variant.cpp
> ```

The latest commit already includes the changes mentioned above, and running the tool locally did not raise any exceptions regarding my changes